cmd/spire-server/cli/run/run.go

```diff
@@ -337,7 +344,15 @@ func NewServerConfig(c *Config, logOptions []log.Option, allowUnknownConfig bool
 	logOptions = append(logOptions,
 		log.WithLevel(c.Server.LogLevel),
 		log.WithFormat(c.Server.LogFormat),
-		log.WithOutputFile(c.Server.LogFile))
+	)
+	if c.Server.LogFile != "" {
+		reopenableFile, err := log.NewReopenableFile(c.Server.LogFile)
+		if err != nil {
+			return nil, err
+		}
+		logOptions = append(logOptions, log.WithReopenableOutputFile(reopenableFile))
+		sc.LogReopener = reopenableFile
+	}
```

Comment:
Agent and Server code seem nearly identical.
Is there a place we could extract the common functionality?

Reply:
Nearly identical, yes, but to extract it to a helper, I think it would need to accept the config under construction as args, allowing one or the other to be nil or as `interface{}`. Neither of those options seems better than similar copypasta as I've done here.